File: api/Entities/Admin/VisaItem.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;

namespace api.Entities.Admin
{
    public class VisaItem: BaseEntity
    {
        public int SrNo { get; set; }
        public int VisaHeaderId { get; set; }
        public string VisaCategoryArabic { get; set; }
        [Required, MinLength(5), MaxLength(50)]
        public string VisaCategoryEnglish { get; set; }
        [Required, MaxLength(15)]
        public string VisaConsulate { get; set; }
        [Required, Range(1,250)]
        public int VisaQuantity { get; set; }
        public bool VisaCanceled { get; set; }=false;
    }
}

File: client/src/app/_models/admin/visaItem.ts
export interface IVisaItem
{
    id: number;
    srNo: number;
    visaHeaderId: number;
    visaNo: string;
    visaCategoryArabic: string;
    visaCategoryEnglish: string;
    visaConsulate: string;
    visaQuantity: string;
    visaCanceled: boolean;
}

export class VisaItem implements IVisaItem
{
    id = 0;
    srNo = 0;
    visaHeaderId = 0;
    visaNo='';
    visaCategoryArabic = '';
    visaCategoryEnglish = '';
    visaConsulate= '';
    visaQuantity = '';
    visaCanceled = false;
}